docs/source/core/adapters/components.rst

```diff
@@ -0,0 +1,90 @@
+
+
+.. autoclass:: nemo.collections.common.parts.adapter_modules.AbstractAdapterModule
+    :show-inheritance:
+    :members:
+    :member-order: bysource
+
+-----
+
+.. autoclass:: nemo.collections.common.parts.adapter_modules.LinearAdapter
+    :show-inheritance:
+    :members:
+    :member-order: bysource
+
+
+Insertion Form - Module Adapters
+--------------------------------
+
+Adapter modules can be integrated into many different locations of a given module. For example, it is possible to have an adapter that affects only the outputs of the final layer in each module. We can also have a ``Parallel Adapter`` :cite:`adapters-Junxian2021unified` that operates at the input of the module itself, in parallel to the forward pass of the module. Yet another insertion location is inside the Multi Head Attention Layers.
+
+On top of this, while adapters are commonly used only in the layers containing the most parameters (say the Encoder of a network), some models can support adapters in multiple locations (Encoder-Decoder architecture for Language Models, Machine Translation, or even Encoder-Decoder-Joint for ASR with Transducer Loss). As such, NeMo utilizes the concept of ``Module Adapters``.
+
+``Module Adapters`` are very simply defined when adding an adapter - by specifying the module that the adapter should be inserted into.
+
+.. code-block:: python
+
+    # Get the list of supported modules / locations in a adapter compatible Model
+    print(model.adapter_module_names)  # assume ['', 'encoder', 'decoder']
+
+    # When calling add_adapter, specify the module name in the left of the colon symbol, and the adapter name afterwords.
+    # The adapter is then directed to the decoder module instead of the default / encoder module.
+    model.add_adapter("decoder:first_adapter", cfg=...)
+
+You might note that ``model.adapter_module_names`` can sometimes return ``''`` as one of the supported module names - this refers to the "default module". Generally we try to provide the default as the most commonly used adapter in literature - for example, Encoder adapters in NLP/NMT/ASR.
+
+Composition Function - Adapter Strategies
+-----------------------------------------
+
+Finally, we discuss how to compose the input and output of adapter modules. In order to generalize this step, we construct ``Adapter Strategies``.
+A strategy is any class (not torch.nn.Module!) that extends :class:`~nemo.core.classes.mixins.adapter_mixin_strategies.AbstractAdapterStrategy`, and provides a ``forward()`` method that accepts a specific signature of the inputs and produces an output tensor which combines the input and output with some specific method.
+
+We discuss a simple residual additional connection strategy below - that accepts an input to the adapter and an adapters output and simply adds them together. It also supports ``stochastic_depth`` which enables adapters to be dynamically switched off during training, making training more robust.
+
```

Comment:
same as above, consider compacting/removing api blocks below.

Reply:
They are necessary to understand what the abstract and implementation of the strategy look like.